include/core/truthTable/truth_table.hpp

```diff
@@ -105,6 +105,48 @@ namespace syrec {
                 return result;
             }
 
+            // return bool vec representation of the cube (order is b0,b1,b2......bn)
+            [[nodiscard]] auto toBoolVec() const -> std::vector<bool> {
+                assert(cube.size() <= 64U);
+                assert(std::none_of(cube.cbegin(), cube.cend(), [](auto const& v) { return !v.has_value(); }));
+                std::vector<bool> result;
+                result.reserve(size());
+                for (auto i = static_cast<int>(size()) - 1; i >= 0; i--) {
+                    result.emplace_back(*cube[i]);
+                }
+                return result;
+            }
+
+            // return string representation of the cube
+            [[nodiscard]] auto toString() const -> std::string {
+                assert(cube.size() <= 64U);
+                std::string result;
+                result.reserve(size());
+                for (auto const& i: cube) {
+                    if (!i.has_value()) {
+                        result.push_back('-');
+                    } else {
+                        const auto valueChar = *i ? '1' : '0';
+                        result.push_back(valueChar);
+                    }
+                }
+                return result;
+            }
+
+            // checks if 2 strings are equal irrespective of don't care
+            [[nodiscard]] static auto equalDcCube(const std::string& in, const std::string& out) -> bool {
+                for (auto i = 0U; i < in.size(); i++) {
+                    if (in[i] == out[i]) {
+                        continue;
+                    }
+                    if (out[i] == '-') {
+                        continue;
+                    }
+                    return false;
+                }
+                return true;
+            }
```

Comment:
Why isn't this function symmetric? And why is there no check on the lengths of both strings? If I were to write a function like that it would probably look something like
```c++
[[nodiscard]] static auto checkCubeEquality(const std::string& c1, const std::string& c2, const bool equalityUpToDontCare = true) -> bool {
  if (c1.size() != c2.size()) {
    return false;
  }
  const auto nBits = in.size();
  for (auto i = 0U; i < nBits; ++i) {
    if (equalityUpToDon'tCare && (c1[i] == '-' || c2[i] == '-')) { 
      continue;
    }
    if (c1[i] != c2[i]) {
      return false;
    }
  }
  return true;
}
```

However, as I identified later after looking at the unittests. That's not even the main problem. There should not be a function that just takes two strings, but rather one that directly works on the Cubes themselves.
In the end, there should be a function (following a similar blueprint as above) that compares two `Cube`s and compares them. In addition, there should be a function that compares two `TruthTable`s using the previous function.

Reply:
👍